lib/stitches/fill.py

```diff
@@ -90,7 +92,26 @@ def stitch_row(stitches, beg, end, angle, row_spacing, max_stitch_length, stagge
         stitches.append(end)
 
 
-def intersect_region_with_grating(shape, angle, row_spacing, end_row_spacing=None, flip=False):
+def intersect_region_with_grating(shape: Shape,
+                                  angle: float,
+                                  row_spacing: int,
+                                  end_row_spacing: int = None,
+                                  flip=False) -> List[RowSegments]:
+    """Intersects the given region with a grating.
+
+    :param shape: The shape to intersect (a single polygon).
+    :param angle: The angle of the grating
+    :param row_spacing: The spacing of the rows
+    :param end_row_spacing: ???  FIXME: interpret
```

Comment:
Two remaining questions are about a description of this parameter and the `flip` parameter below (if there's a good explanation of why it's needed).

Reply:
End row spacing defines the spacing of rows at the end of the shape. This gives us the possibility to have different row spacings at the beginning and the end of a shape. Very useful if you want to do color blending (https://inkstitch.org/docs/features/#color-blending)

The flip parameter is only interesting for the legacy (manual) fill stitch and helps to route these type of stitching (https://inkstitch.org/docs/params/#legacy-fill-params)

Reply:
Thanks! Fixed